src/addagent/manage_agents.c

```diff
@@ -98,14 +106,26 @@ int add_agent(int json_output)
     c_ip.ip = NULL;
 
     char *id_exist = NULL;
-    int force_antiquity = INT_MAX;
+    authd_force_options_t authd_force_options = {0};
     int sock;
     int authd_running;
 
-    const char *env_remove_dup = getenv("OSSEC_REMOVE_DUPLICATED");
+    // Creating the configuration structure according to the parameters used
+    const char *env_disconnected_time = getenv("DISCONNECTED_TIME");
 
-    if (env_remove_dup) {
-        force_antiquity = strtol(env_remove_dup, NULL, 10);
+    if (env_disconnected_time) {
+        authd_force_options.disconnected_time = strtol(env_disconnected_time, NULL, 10);
+        authd_force_options.disconnected_time_enabled = true;
+        authd_force_options.enabled = true;
+    } else{
+        authd_force_options.disconnected_time_enabled = false;
```

Comment:
As we are assuming **authd_force_options.enabled** is disabled by default, we can avoid this **else** statement to keep the same logic.

Reply:
Fixed!